hedera-node/hedera-app/src/test/java/com/hedera/node/app/state/merkle/disk/OnDiskTest.java

```diff
@@ -0,0 +1,258 @@
+/*
+ * Copyright (C) 2022 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hedera.node.app.state.merkle.disk;
+
+import com.hedera.node.app.state.merkle.MerkleTestBase;
+import com.hedera.node.app.state.merkle.StateMetadata;
+import com.swirlds.common.merkle.crypto.MerkleCryptoFactory;
+import com.swirlds.common.merkle.crypto.MerkleCryptography;
+import com.swirlds.virtualmap.VirtualMap;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import org.junit.jupiter.api.io.TempDir;
+
+/**
+ * A variety of robust tests for the on-disk merkle data structure, especially including
+ * serialization to/from disk (under normal operation) and to/from saved state. These tests use a
+ * more complex map, with full objects to store and retrieve objects from the virtual map, and when
+ * serializing for hashing, and for serializing when saving state.
+ */
+class OnDiskTest extends MerkleTestBase {
+    private static final String SERVICE_NAME = "CryptoService";
+    private static final String ACCOUNT_STATE_KEY = "Account";
+    private static final MerkleCryptography CRYPTO = MerkleCryptoFactory.getInstance();
+
+    private StateMetadata<AccountID, Account> md;
+    private VirtualMap<OnDiskKey<AccountID>, OnDiskValue<Account>> virtualMap;
+    @TempDir Path storageDir;
+
+    //    @Override
+    //    @BeforeEach
+    //    protected void setUp() {
+    //        super.setUp();
+    //        md =
+    //                new StateMetadata<>(
+    //                        SERVICE_NAME,
```

Comment:
Can this be deleted

Reply:
And now I have uncommented it and made it pass again. Yay!